### PR TITLE
[HOPSWORKS-1769] enable publishing models to other projects

### PR DESCRIPTION
**Pull request by o-alex**

### CODE REVIEW

File: hops/experiment_impl/util/experiment_utils.py [hunk 1]
```diff
@@ -443,8 +442,9 @@ def _attach_experiment_xattr(app_id, run_id, json_data, xattr):
                    constants.REST_CONFIG.HOPSWORKS_PROJECT_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER + \
                    hdfs.project_id() + constants.DELIMITERS.SLASH_DELIMITER + \
                    constants.REST_CONFIG.HOPSWORKS_EXPERIMENTS_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER + \
-                   app_id + "_" + str(run_id) + "?xattr=" + xattr
+                   ml_id + "?type=" + op_type
 
+    print("experiment xattr" + resource_url + " body:" + json_data)
```
Comment (robzor92) [inline hops/experiment_impl/util/experiment_utils.py:447]: Leftover from debugging?

File: hops/experiment_impl/util/experiment_utils.py [hunk 2]
```diff
@@ -501,6 +476,7 @@ def _attach_model_xattr(ml_id, json_data):
                    constants.REST_CONFIG.HOPSWORKS_MODELS_RESOURCE + constants.DELIMITERS.SLASH_DELIMITER + \
                    ml_id
 
+    print("model xattr:" + resource_url + " body:" + json_data)
```
Comment (robzor92) [inline hops/experiment_impl/util/experiment_utils.py:479]: leftover?

File: hops/hdfs.py [hunk 3]
```diff
@@ -87,6 +88,20 @@ def project_id():
     """
     return os.environ[constants.ENV_VARIABLES.HOPSWORKS_PROJECT_ID_ENV_VAR]
 
+def project_id_as_shared(name=None):
```
Comment (robzor92) [inline hops/hdfs.py:91]: Maybe we can put this in the project.py module instead?

Comment (robzor92) [reply hops/hdfs.py:91]: Or if this is mainly for internal use we can put it in util.py

**Review by robzor92 — CHANGES_REQUESTED** [review]

File: hops/experiment_impl/util/experiment_utils.py (hunk 1 above)
Comment (o-alex) [reply hops/experiment_impl/util/experiment_utils.py:447]: fixed

File: hops/experiment_impl/util/experiment_utils.py (hunk 2 above)
Comment (o-alex) [reply hops/experiment_impl/util/experiment_utils.py:479]: fixed

File: hops/hdfs.py (hunk 3 above)
Comment (o-alex) [reply hops/hdfs.py:91]: moved to project

**Review by robzor92 — APPROVED** [review]

**Pull request merged**